src/controllers/handlers/resize-handler.ts

```diff
@@ -0,0 +1,66 @@
+import { HandlerContextWithPath } from '../../types'
+
+import createCommandTrigger from '../../adapters/commandLine'
+
+export async function resizeHandler(
+  context: HandlerContextWithPath<
+    'storages' | 'conversionBuilder' | 'assetRetriever' | 'config' | 'logs',
+    '/content/:hash/dxt/:length'
+  >
+) {
+  const {
+    components: { storages, conversionBuilder, assetRetriever, config, logs },
+    params
+  } = context
+  const logger = logs.getLogger('resize-handler')
+
+  const assetHash = params.hash
+  const lengthRequired = params.length
+
+  if (!assetHash || !lengthRequired) return { status: 400 }
+
+  logger.info(`Processing with ${JSON.stringify({ assetHash, lengthRequired })}`)
+
+  const originalAssetName = `${assetHash}.png`
+  const convertedAssetName = `${assetHash}.crn`
+  const assetToUploadName = `${assetHash}-${lengthRequired}.crn`
+
+  if (!(await storages.bucket.exist(assetToUploadName))) {
+    const asset: ArrayBuffer = await assetRetriever(assetHash)
+
+    if (!asset) {
+      return { status: 404 }
+    }
+
+    const filePath = storages.local.saveFile(originalAssetName, asset)
+
+    const conversionProcesss = conversionBuilder
+      .setDestination(convertedAssetName)
+      .withAdjustedSize(Number(lengthRequired))
+      .buildConversion(filePath, 'crn')
+      .executeWith(createCommandTrigger())
+
+    conversionProcesss.onError((data: Buffer) => {
+      logger.error(`Process failed with: ${data.toString()}`)
+    })
+
+    const conversionResult: { code: number; failed: boolean } = await new Promise((resolve) => {
+      conversionProcesss.onEnd((code: number) => {
+        logger.info(`Process finished with code ${code}`)
+        resolve({ code, failed: code != 0 })
+      })
+    })
+    storages.local.deleteFile(originalAssetName)
+
+    if (conversionResult.failed) {
+      return { status: 400 }
+    }
+
+    await storages.bucket.storeStream(assetToUploadName, storages.local.asReadable(convertedAssetName))
+    storages.local.deleteFile(convertedAssetName)
```

Comment:
```suggestion
    await storages.local.deleteFile(convertedAssetName)
```

Reply:
This can be handled by the "floating promises" eslint rule, it is extremely important not to ignore it because unhandled rejections will kill the server

Reply:
I'm using unlinkSync to remove these files that's why I'm not awaiting, anyways I'll change it to do it async, thanks